src/actions/profile-view.js

```diff
@@ -338,15 +338,13 @@ export function changeRightClickedTrack(
 }
 
 export function setContextMenuVisibility(
-  isVisible: boolean
-): ThunkAction<void> {
-  return (dispatch, getState) => {
-    const selectedThreadIndex = getSelectedThreadIndex(getState());
-    dispatch({
-      type: 'SET_CONTEXT_MENU_VISIBILITY',
-      threadIndex: selectedThreadIndex,
-      isVisible,
-    });
+  isVisible: boolean,
+  threadIndex: ThreadIndex
+): Action {
+  return {
+    type: 'SET_CONTEXT_MENU_VISIBILITY',
+    threadIndex,
+    isVisible,
   };
```

Comment:
While I worked on this some weeks ago, I thought that the concept of "context menu visibility" and "right clicked marker" and "right clicked call node path" shouldn't be tied to a thread.

The idea would be to keep the state of "right clicked marker" and "right clicked call node path" out of `viewOptionsPerThread` as it is now, in their own reducers. After all there can be only right clicked value at any time. Of course the threadIndex would be part of these states (a marker index or a call node path has a meaning with its thread index only).

And so the `threadIndex` here would be useless, as well as a lot of the changes you had to make so that we could pass it here.

Such a change would mean changing the reducers.